internal/app/api/balance/handler.go

```diff
@@ -0,0 +1,168 @@
+package balance
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"github.com/go-chi/chi/v5"
+	"github.com/korol8484/gofermart/internal/app/api/util"
+	"github.com/korol8484/gofermart/internal/app/domain"
+	"go.uber.org/zap"
+	"io"
+	"net/http"
+	"time"
+)
+
+type listResponse struct {
+	Order       string  `json:"order"`
+	Sum         float64 `json:"sum"`
+	ProcessedAt string  `json:"processed_at"`
+}
+
+type balanceResponse struct {
+	Current float64 `json:"current"`
+	Sum     float64 `json:"withdrawn"`
+}
+
+type withdrawRequest struct {
+	Order string  `json:"order"`
+	Sum   float64 `json:"sum"`
+}
+
+type usecase interface {
+	UserWithdrawals(ctx context.Context, userID domain.UserID) ([]*domain.Balance, error)
+	GetUserSumWC(ctx context.Context, userID domain.UserID) (*domain.SumWC, error)
```

Comment:
Мне кажется, что это не совсем удачный нейминг для метода. В названии есть информация GetUser, которой как раз как мне кажется можно пренебречь, но то, что именно делает/возвращает метод сокращено до аббревиатуры WC.
```suggestion
	GetUserSumWC(ctx context.Context, userID domain.UserID) (*domain.SumWC, error)
```